molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/controller/DataExplorerController.java

```diff
@@ -363,15 +365,33 @@ public void download(@RequestParam("dataRequest") String dataRequestStr, HttpSer
 				download.writeToCsv(dataRequest, outputStream, ',');
 				break;
 			case DOWNLOAD_TYPE_XLSX:
-				response.setContentType("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet");
-				response.addHeader("Content-Disposition", "attachment; filename=\"" + fileName + "\"");
 
-				outputStream = response.getOutputStream();
-				download.writeToExcel(dataRequest, outputStream);
+				if (getNumberOfCellsForEntityType(dataRequest.getEntityName(), dataRequest.getQuery())
```

Comment:
Move to DataExplorerDownloadHandler, since we don't want other code paths using DataExplorerDownloadHandler to bring down the system on larger downloads.